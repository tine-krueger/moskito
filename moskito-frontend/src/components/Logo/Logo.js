
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import moskitoLogo from '../../images/moskitoLogo.svg'

Logo.protoTypes = { inputMargin: PropTypes.string }

export default function Logo({inputMargin}) {
    return (
        <LogoStyled inputMargin={inputMargin}>
                <LogoImage src={moskitoLogo} alt="Moskito Logo"/>
        </LogoStyled>
    )
}

const LogoStyled = styled.div`
    margin-top: ${props => props.inputMargin || '8em'};
    text-align: center;
`

const LogoImage = styled.img`
    box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2), 
                inset -3px -4px 6px 0 rgba(255,255,255,0.3);
    border-radius: 20px;
    padding: 1em 2em;
    width: 85%;
`
