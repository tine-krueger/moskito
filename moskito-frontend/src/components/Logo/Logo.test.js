import { render } from '@testing-library/react'
import 'jest-styled-components'
import Logo from './Logo'

const inputMargin = '2em'

describe('Logo', () => {

    it('renders correctly', () => {
        const { container } = render(<Logo/>)
        expect(container).toMatchSnapshot()
    })

    it('has margin-top of 4em, of not inputMargin given', () => {
        const { getByTestId } = render(<Logo/>)
        expect(getByTestId('logo')).toHaveStyle('margin-top: 4em')
    })

    it('has margin-top of 2em, with given inputMargin of 2em', () => {
        const { getByTestId } = render(<Logo inputMargin={inputMargin}/>)
        expect(getByTestId('logo')).toHaveStyle('margin-top: 2em')
    })
})