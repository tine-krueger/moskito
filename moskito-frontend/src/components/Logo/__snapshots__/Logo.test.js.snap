// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders correctly 1`] = `
.c0 {
  margin-top: 4em;
  text-align: center;
}

.c1 {
  box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2),inset -3px -4px 6px 0 rgba(255,255,255,0.3);
  border-radius: 20px;
  padding: 1em 2em;
  width: 85%;
}

<div>
  <div
    class="c0"
    data-testid="logo"
  >
    <img
      alt="Moskito Logo"
      class="c1"
      src="moskitoLogo.svg"
    />
  </div>
</div>
`;
